src/gennylib/include/gennylib/conventions.hpp

```diff
@@ -124,12 +126,91 @@ inline bool operator==(const RateSpec& lhs, const RateSpec& rhs) {
     return (lhs.per == rhs.per) && (lhs.operations == rhs.operations);
 }
 
+struct PhaseRangeSpec {
+    PhaseRangeSpec() = default;
+    ~PhaseRangeSpec() = default;
+
+    PhaseRangeSpec(genny::IntegerSpec s, genny::IntegerSpec e) {
+        assert(s.value >= 0 && e.value >= 0 && s.value <= UINT_MAX && e.value <= UINT_MAX && s.value <= e.value);
+        start = static_cast<genny::PhaseNumber>(s.value);
+        end = static_cast<genny::PhaseNumber>(e.value);
+    }
```

Comment:
you can make the assignment into initialization:

```c++
PhaseRangeSpec(IntSpec s, IntSpec e)
: start{static_cast<PhaseNumber>(s)}, end{...} {
  assert(...)
}
```

Moreover I think you want to throw an `InvalidConfigurationException` rather than `assert` since `assert`s will get compiled away.

Reply:
Done